app/interactors/outdated_prolem_clearer.rb

```diff
@@ -0,0 +1,32 @@
+require 'problem_destroy'
+
+class OutdatedProblemClearer
+
+  ##
+  # Clear all problem not present for more than one week.
+  #
+  def execute
+    nb_problem_outdated.tap { |nb|
+      if nb > 0
+        criteria.each do |problem|
+          ProblemDestroy.new(problem).execute
+        end
+        repair_database
+      end
+    }
+  end
+
+  private
+
+  def nb_problem_outdated
+    @count ||= criteria.count
+  end
+
+  def criteria
+    @criteria ||= Problem.outdated
+  end
+
+  def repair_database
+    Mongoid.default_session.command :repairDatabase => 1
```

Comment:
This is probably fine. I can't remember why or if this is needed, but I see we have it elsewhere so I guess we probably should do exactly what you have done unless we figure out that answer.